### PR TITLE
Disable formatting on save for Dart and HTML files

### LINKED ISSUE #122050: Disable `dart format` on save for VSCode contributors
We should add a default workspace settings file for VSCode (one of the primary editors used by Flutter contributors) to ensure that it doesn't format on save by default while contributors are making changes to the Flutter repository. 

It's frustratingly common that someone makes a good change, saves it, and then realizes to their horror that they have reformatted the entire class file. They then have to go through and revert the formatting changes while retaining the semantic changes they've made, a time-consuming and somewhat error-prone process. Adding this file to override the default VSCode settings will remove one tripwire for new contributors to the project.

### PR DESCRIPTION
**Pull request by timsneath**
Adds a default workspace settings file for VSCode (one of the primary editors used by Flutter contributors) to ensure that it doesn't format on save by default while contributors are making changes to the Flutter repository. 

It's frustratingly common that someone makes a good change, saves it, and then realizes to their horror that they have reformatted the entire class file. They then have to go through and revert the formatting changes while retaining the semantic changes they've made, a time-consuming and somewhat error-prone process. Adding this file to override the default VSCode settings will remove one tripwire for new contributors to the project.

Fixes https://github.com/flutter/flutter/issues/122050. This change should be test-exempt since it only adds a configuration file. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
test-exempt: configuration change

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// VSCode workspace settings that are shared among all users of this project.
+// This only affects subdirectories of this project.
+{
+    // VSCode formats files on save by default. Since Flutter source code is
+    // hand-formatted, the default setting is changed to prevent inadvertent
+    // reformatting of code.
+    "editor.formatOnSave": false,
```
Comment (DanTup) [inline .vscode/settings.json:7]: I would suggest also adding:

```
	"editor.formatOnSave": true,
	"editor.formatOnType": true,
```

**Review by DanTup — APPROVED** [review]
LGTM with a suggestion.

FWIW, I think this will only apply if you open up the repo root, and not just some individual packages (as I do, although I use a `.code-workspace` file that disables these).

I also think these aren't defaults in VS Code, but we do have them as "recommended settings" for Dart (so if you accept the prompt to use recommended settings on first run, they'll be added to your User Settings in the `"[dart]"` section). If we think this is the wrong thing, we can change it.

(it would perhaps also be nice if this could be set in a non-VS-Code specific way, although I don't have a good place in mind)

File: .vscode/settings.json (hunk 1 above)
Comment (DanTup) [reply .vscode/settings.json:7]: Whoops, yes! I copy/pasted from my actual settings 😄

Comment (timsneath) [reply .vscode/settings.json:7]: OK, just checking!

**Pull request merged**